packages/node-app/src/tests/giveaway.controller.test.ts

```diff
@@ -471,4 +477,48 @@ describe('PUT giveaways/:id/participants/:participantId', () => {
       expect(giveawaysContract.methods.addParticipant).not.toHaveBeenCalled();
     })
   );
+
+  it(
+    'should not update state if user is not admin',
+    authenticated(async () => {
+      const giveaway = await Giveaway.create({
+        ...giveawayData,
+        startTime: Date.now() + 60,
+        endTime: Date.now() + 10000,
+      });
+      await wait(200);
+
+      const body = { state: ParticipantState.REJECTED };
+      const res = await request(app)
+        .put(`/giveaways/${giveaway._id}/participants/${participant.id}`)
+        .send(body);
+
+      const updatedGiveaway = await Giveaway.findById(giveaway._id);
+      expect(res.status).toEqual(401);
+      expect(updatedGiveaway.participants[0].state).toEqual(
+        ParticipantState.PENDING
+      );
+    })
+  );
+
+  it(
+    'should not update other user notified flag',
+    authenticated(async () => {
+      const giveaway = await Giveaway.create({
+        ...giveawayData,
+        startTime: Date.now() + 60,
+        endTime: Date.now() + 10000,
+      });
+      await wait(200);
+
+      const body = { notified: true };
+      const res = await request(app)
+        .put(`/giveaways/${giveaway._id}/participants/${participant.id}`)
+        .send(body);
+
+      const updatedGiveaway = await Giveaway.findById(giveaway._id);
+      expect(res.status).toEqual(401);
+      expect(updatedGiveaway.participants[0].notified).toEqual(false);
```

Comment:
Chai has a bunch of cool assertions if you ever need them: https://www.chaijs.com/api/bdd/
```suggestion
      expect(updatedGiveaway.participants[0].notified).to.be.false;
```

Reply:
In this specific case I don't the benefit of adding one more library just to achieve the same thing but with a different syntax. We would have to use expect method from chai which would create syntax conflict with the assertions that already exist

Reply:
Sorry, forgot I only installed chai on the truffle app